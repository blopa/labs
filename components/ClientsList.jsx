import { memo } from 'react';
import { List } from '@mui/material';
import PropTypes from 'prop-types';

// Components
import ClientListItem from './ClientListItem';

const propTypes = {
    clients: PropTypes.arrayOf(
        PropTypes.shape({
            name: PropTypes.string.isRequired,
            title: PropTypes.string.isRequired,
            avatar: PropTypes.string.isRequired,
            quote: PropTypes.string,
            nationality: PropTypes.string,
        }).isRequired
    ).isRequired,
};

function ClientsList({ clients }) {
    return (
        <List
            sx={{ width: '100%', maxWidth: 360, bgcolor: 'background.paper' }}
        >
            {clients.map((client, index) => {
                return <ClientListItem key={index} client={client} />;
            })}
        </List>
    );
}

ClientsList.propTypes = propTypes;

export default memo(ClientsList);
