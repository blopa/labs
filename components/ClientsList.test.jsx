import { render } from '@testing-library/react';
import ClientsList from './ClientsList';

it('renders correctly', () => {
    const clients = [
        {
            name: 'Ash Ketchum',
            title: 'Pokémon World Champion',
            avatar: 'https://avataaars.io/?avatarStyle=Circle&topType=ShortHairDreads01&accessoriesType=Blank&hairColor=BrownDark&facialHairType=Blank&clotheType=BlazerShirt&eyeType=Default&eyebrowType=Default&mouthType=Smile&skinColor=Light',
            quote: 'I wanna be the very best.',
            nationality: 'Kantonian',
        },
        {
            name: 'Misty',
            title: 'Water Type Pokémon Trainer',
            avatar: 'https://avataaars.io/?avatarStyle=Circle&topType=LongHairShavedSides&accessoriesType=Blank&facialHairType=Blank&clotheType=BlazerShirt&eyeType=Default&eyebrowType=Default&mouthType=Default&skinColor=Light',
            quote: 'I am Misty, the tomboyish mermaid!',
            nationality: 'Kantonian',
        },
    ];

    const { container } = render(<ClientsList clients={clients} />);

    expect(container).toMatchSnapshot();
});
