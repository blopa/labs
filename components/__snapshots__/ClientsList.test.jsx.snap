// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <ul
    class="MuiList-root MuiList-padding css-1mcnwpj-MuiList-root"
  >
    <div
      class="MuiBox-root css-0"
    >
      <div
        class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-16ac5r2-MuiButtonBase-root-MuiListItemButton-root"
        role="button"
        tabindex="0"
      >
        <div
          class="MuiListItemAvatar-root css-1e9lk82-MuiListItemAvatar-root"
        >
          <div
            class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-2s90m6-MuiAvatar-root"
          >
            <img
              alt="Picture of Ash Ketchum"
              data-nimg="1"
              decoding="async"
              height="50"
              loading="lazy"
              src="/_next/image?url=https%3A%2F%2Favataaars.io%2F%3FavatarStyle%3DCircle%26topType%3DShortHairDreads01%26accessoriesType%3DBlank%26hairColor%3DBrownDark%26facialHairType%3DBlank%26clotheType%3DBlazerShirt%26eyeType%3DDefault%26eyebrowType%3DDefault%26mouthType%3DSmile%26skinColor%3DLight&w=128&q=75"
              srcset="/_next/image?url=https%3A%2F%2Favataaars.io%2F%3FavatarStyle%3DCircle%26topType%3DShortHairDreads01%26accessoriesType%3DBlank%26hairColor%3DBrownDark%26facialHairType%3DBlank%26clotheType%3DBlazerShirt%26eyeType%3DDefault%26eyebrowType%3DDefault%26mouthType%3DSmile%26skinColor%3DLight&w=64&q=75 1x, /_next/image?url=https%3A%2F%2Favataaars.io%2F%3FavatarStyle%3DCircle%26topType%3DShortHairDreads01%26accessoriesType%3DBlank%26hairColor%3DBrownDark%26facialHairType%3DBlank%26clotheType%3DBlazerShirt%26eyeType%3DDefault%26eyebrowType%3DDefault%26mouthType%3DSmile%26skinColor%3DLight&w=128&q=75 2x"
              style="color: transparent;"
              width="50"
            />
          </div>
        </div>
        <div
          class="MuiListItemText-root MuiListItemText-multiline css-konndc-MuiListItemText-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
          >
            Ash Ketchum
          </span>
          <p
            class="MuiTypography-root MuiTypography-body2 MuiListItemText-secondary css-83ijpv-MuiTypography-root"
          >
            Pokémon World Champion
          </p>
        </div>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </div>
    </div>
    <div
      class="MuiBox-root css-0"
    >
      <div
        class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-16ac5r2-MuiButtonBase-root-MuiListItemButton-root"
        role="button"
        tabindex="0"
      >
        <div
          class="MuiListItemAvatar-root css-1e9lk82-MuiListItemAvatar-root"
        >
          <div
            class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-2s90m6-MuiAvatar-root"
          >
            <img
              alt="Picture of Misty"
              data-nimg="1"
              decoding="async"
              height="50"
              loading="lazy"
              src="/_next/image?url=https%3A%2F%2Favataaars.io%2F%3FavatarStyle%3DCircle%26topType%3DLongHairShavedSides%26accessoriesType%3DBlank%26facialHairType%3DBlank%26clotheType%3DBlazerShirt%26eyeType%3DDefault%26eyebrowType%3DDefault%26mouthType%3DDefault%26skinColor%3DLight&w=128&q=75"
              srcset="/_next/image?url=https%3A%2F%2Favataaars.io%2F%3FavatarStyle%3DCircle%26topType%3DLongHairShavedSides%26accessoriesType%3DBlank%26facialHairType%3DBlank%26clotheType%3DBlazerShirt%26eyeType%3DDefault%26eyebrowType%3DDefault%26mouthType%3DDefault%26skinColor%3DLight&w=64&q=75 1x, /_next/image?url=https%3A%2F%2Favataaars.io%2F%3FavatarStyle%3DCircle%26topType%3DLongHairShavedSides%26accessoriesType%3DBlank%26facialHairType%3DBlank%26clotheType%3DBlazerShirt%26eyeType%3DDefault%26eyebrowType%3DDefault%26mouthType%3DDefault%26skinColor%3DLight&w=128&q=75 2x"
              style="color: transparent;"
              width="50"
            />
          </div>
        </div>
        <div
          class="MuiListItemText-root MuiListItemText-multiline css-konndc-MuiListItemText-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
          >
            Misty
          </span>
          <p
            class="MuiTypography-root MuiTypography-body2 MuiListItemText-secondary css-83ijpv-MuiTypography-root"
          >
            Water Type Pokémon Trainer
          </p>
        </div>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </div>
    </div>
  </ul>
</div>
`;
